src/server.rs

```diff
@@ -49,16 +50,21 @@ pub async fn start_server(
     for method in &config.rpcs.methods {
         let mut list: Vec<Arc<dyn Middleware<_, _>>> = vec![];
 
-        if let Some(index) = method.inject_block_hash() {
-            list.push(Arc::new(InjectParamsMiddleware::new(
-                api.clone(),
-                InjectType::BlockHashAt(index),
-            )));
-        } else if let Some(index) = method.inject_block_num() {
-            list.push(Arc::new(InjectParamsMiddleware::new(
-                api.clone(),
-                InjectType::BlockNumberAt(index),
-            )));
+        match method.at() {
+            // TODO: inclue params into middleware and do this inside
```

Comment:
need to do in this PR?

Reply:
This is a PR to your branch. I wanted to give this idea and let you finish the feature

Reply:
I'm not fond of this idea very much tbh. I'm not against it merged to my branch if everyone else is happy, but that's all. You should finish your idea/work.

Reply:
I'm closing this, let's keep it simple